### DISCUSSION TITLE
Proxy for Database

### DISCUSSION THREAD

**Discussion by douglasoliveiraadv** [nginx-proxy/nginx-proxy#2490] [Ideas]
How can I make the proxy serve mysql or firebird? My Firebird database is in a container on port 3050, and that port just does not work. I knew the nginx-stream project but I can not make it work in the same way as nginx-proxy.

Please help me!

**Comment by douglasoliveiraadv** [comment]
I tried to apply as changes from https://github.com/jmparra/nginx-proxy/commit/b3977a8f335dda0a3285c0e435074f4e3bf423d4#diff-5bd6b85c2d6fc987875f4bf82de2a15a in the current version of jwilder/nginx-proxy, but it did not work.

**Comment by douglasoliveiraadv** [comment]
I also tried with this https://github.com/drud/docker.ddev-router, but also did not work.

**Comment by dextervip** [comment]
Any workaround?

**Comment by pkimtani** [comment]
I had a similar situation and following actually worked for me:

/etc/nginx/nginx.conf
```
...
stream {
    upstream db.localhost-1 {
        server <ip_address_of_container>:3306;
    }
    server {
        listen 3306;
        proxy_pass db.localhost-1;
    }
}
...
```

I couldn't find a more versatile or robust solution using the options in this package so I created by own nginx.conf and mounted it to the container. May be someday I can open a PR about this :)

Hope it helps someone :)

**Comment by victorlunam** [comment]
Does anyone know if there are plans to support proxy support for database connections ?

**Reply by buchdag** [reply]
There isn't really any plans to support `stream` proxying at the moment.

**Reply by victorlunam** [reply]
One of the most important reasons to use nginx-proxy, is because it is easy to configure (the simplest way, create an instance of the proxy and link the other containers with VIRTUAL_HOST for HTTP connections).

The way to generate a new configuration automatically is just great.

But it stops when we want to add support for TCP and UDP connections.

 We have some options like: 
* Difficult: edit and/or link the /etc/nginx/nginx.conf file.
* Easy: migrate to other alternatives ...
* Easy: export connections on different ports.... :/ (although not really a solution, but it is very common)

I think the most common use of TCP/UDP proxy is database connections.

I think it would be an important feature to add this support for a next version of nginx-proxy.
